### PR TITLE
Use JksOptions instead of raw JsonObject

### PR DESCRIPTION
**Pull request by pmlopes**
As discussed before, replace raw JsonObject config with JksOptions

### CODE REVIEW

**Comment by pmlopes** [comment]
@vietj please review

File: vertx-auth-jwt/src/main/generated/io/vertx/rxjava/ext/auth/jwt/JWTAuth.java [hunk 1]
```diff
@@ -50,12 +51,32 @@ public Object getDelegate() {
    * @param config the config
    * @return the auth provider
    */
-  public static JWTAuth create(Vertx vertx, JsonObject config) { 
+  public static JWTAuth create(Vertx vertx, JksOptions config) {
```
Comment (cescoffier) [inline vertx-auth-jwt/src/main/generated/io/vertx/rxjava/ext/auth/jwt/JWTAuth.java:54]: This is breaking the API. The method should be kept and marked as deprecated.

File: vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/JWTAuth.java [hunk 2]
```diff
@@ -37,11 +39,30 @@
    * @param config  the config
    * @return the auth provider
    */
-  static JWTAuth create(Vertx vertx, JsonObject config) {
+  static JWTAuth create(Vertx vertx, JksOptions config) {
```
Comment (cescoffier) [inline vertx-auth-jwt/src/main/java/io/vertx/ext/auth/jwt/JWTAuth.java:42]: This is breaking the API. The method should be kept and marked as deprecated.

I'm actually not sure we can just do a new JksOptions(config) ?

File: vertx-auth-jwt/src/test/java/io/vertx/ext/auth/test/jwt/JWTAuthProviderTest.java [hunk 3]
```diff
@@ -38,11 +39,10 @@ public void setUp() throws Exception {
     authProvider = JWTAuth.create(vertx, getConfig());
   }
 
-  protected JsonObject getConfig() {
```
Comment (cescoffier) [inline vertx-auth-jwt/src/test/java/io/vertx/ext/auth/test/jwt/JWTAuthProviderTest.java:41]: A test with the two way to create the authenticator would be nice (using JKSOptions and JsonObject)

**Comment by pmlopes** [comment]
Since this would break the current API 3.x it will be postponed

**Pull request closed without merging**